x/evm/types/msg.go

```diff
@@ -123,13 +90,13 @@ func (msg MsgEthereumTx) Type() string { return TypeMsgEthereumTx }
 // ValidateBasic implements the sdk.Msg interface. It performs basic validation
 // checks of a Transaction. If returns an error if validation fails.
 func (msg MsgEthereumTx) ValidateBasic() error {
-	gasPrice := new(big.Int).SetBytes(msg.Data.GasPrice)
+	gasPrice := msg.Data.gasPrice()
 	if gasPrice.Sign() == -1 {
 		return sdkerrors.Wrapf(ErrInvalidGasPrice, "gas price cannot be negative %s", gasPrice)
 	}
 
 	// Amount can be 0
-	amount := new(big.Int).SetBytes(msg.Data.Amount)
+	amount := msg.Data.amount()
 	if amount.Sign() == -1 {
```

Comment:
check for amount not null otherwise it panics (currently it fails when deplying contract)